modules/core/src/smithy4s/schema/EnumTag.scala

```diff
@@ -16,28 +16,26 @@
 
 package smithy4s.schema
 
-sealed trait EnumTag[+E]
+// scalafmt: { maxColumn = 120, align.preset = most}
+sealed trait EnumTag[E]
 
 object EnumTag {
-  case object ClosedStringEnum extends EnumTag[Nothing]
-  case object ClosedIntEnum extends EnumTag[Nothing]
-  case class OpenStringEnum[E](unknown: String => E) extends EnumTag[E]
-  case class OpenIntEnum[E](unknown: Int => E) extends EnumTag[E]
+  case class StringEnum[E](value: E => String, unknown: Option[String => E]) extends EnumTag[E]
+  case class IntEnum[E](value: E => Int, unknown: Option[Int => E])          extends EnumTag[E]
 
-  object StringEnum {
-    def unapply[E](enumTag: EnumTag[E]): Boolean = enumTag match {
-      case ClosedStringEnum  => true
-      case OpenStringEnum(_) => true
-      case _                 => false
+  object OpenStringEnum {
```

Comment:
I'm probably gonna remove these : these custom extractors are getting in the way of exhaustive pat-mats